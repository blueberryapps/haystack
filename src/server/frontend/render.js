import Helmet from 'react-helmet';
import React from 'react';
import ReactDOMServer from 'react-dom/server';
import createStore from '../../common/createStore';
import Html from './Html.react';
import ServerProvider from './ServerProvider.react';

export default function render(req, app, options = {}) {
  const store = createStore({}, {});
  const appHtml = ReactDOMServer.renderToString(
    <ServerProvider
      url={req.url}
      context={options.staticContext || {}}
      store={store}
    >
      {app}
    </ServerProvider>
  );

  const { javascript: javascripts } = webpackIsomorphicTools.assets();

  const docHtml = ReactDOMServer.renderToStaticMarkup(
    <Html
      javascripts={javascripts}
      options={options}
      helmet={Helmet.rewind()}
      bodyHtml={`<div id="app">${appHtml}</div>`}
      reduxState={store.getState()}
    />
  );
  return `<!DOCTYPE html>${docHtml}`;
}
