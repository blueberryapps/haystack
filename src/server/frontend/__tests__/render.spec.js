import React from 'react';
import Helmet from 'react-helmet';
import render from '../render';

test('render', () => {
  global.webpackIsomorphicTools = { assets: () => ({ javascript: { app: 'app.xxx.js' } }) };
  Helmet.canUseDOM = false;
  expect(render(<div />)).toMatchSnapshot();
});
