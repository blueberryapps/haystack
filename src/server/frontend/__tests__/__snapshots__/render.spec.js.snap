exports[`test render 1`] = `"<!DOCTYPE html><html lang=\"en\"><head><meta charset=\"utf-8\"/><meta content=\"width=device-width, initial-scale=1, shrink-to-fit=no, user-scalable=no\" name=\"viewport\"/><meta content=\"ie=edge\" http-equiv=\"x-ua-compatible\"/><title data-react-helmet=\"true\"></title></head><body><div id=\"app\"><div id=\"app\"><div data-reactroot=\"\" data-reactid=\"1\" data-react-checksum=\"1998851930\"></div></div></div><script src=\"app.xxx.js\" type=\"text/javascript\"></script></body></html>"`;
